docs.feldera.com/docs/pipelines/latency.md

```diff
@@ -0,0 +1,211 @@
+metadata that helps the user to measure this latency. This per-connector metadata is only
+available for certain connector types,
+via the [`/stats`](https://docs.feldera.com/api/retrieve-statistics-e-g-performance-counters-of-a-running-or-paused-pipeline)
+endpoint in the `.inputs[<connector_index>].completed_frontier` field.
+It describes the latest offset in the input data stream fully processed by the pipeline.
+It includes the following fields:
+
+| Field         | Description                                                                                   | Format                                |
+|---------------|-----------------------------------------------------------------------------------------------|---------------------------------------|
+| `metadata`    | Connector-specific metadata that describes the latest offset in the input stream fully processed by the pipeline (e.g., Kafka partition/offset pairs). | connector-specific (see below)|
+| `ingested_at` | Timestamp when data was ingested from the wire.                                               | RFC 3339 timestamp                    |
+| `processed_at`| Timestamp when the SQL engine finished processing it.                                         | RFC 3339 timestamp                    |
+| `completed_at`| Timestamp when all resulting outputs were delivered to all sinks.                             | RFC 3339 timestamp                    |
+
+The `completed_frontier` property is currently supported for the following connectors:
+* [Kafka input connector](/connectors/sources/kafka)
+* [Delta Lake connector](/connectors/sources/delta)
+
+These connectors support the following metadata formats:
+
+<Tabs>
+    <TabItem value="delta" label="Delta Lake">
+    ```json
+    {
+        "version": N
+    }
+    ```
+    where `version` is the latest Delta table version processed by the pipeline.
+    </TabItem>
+
+    <TabItem value="kafka" label="Kafka">
+    ```json
+    {
+        "offsets": [
+        {
+            "start": START_OFFSET1,
+            "end": END_OFFSET1
+        },
+        {
+            "start": START_OFFSET2,
+            "end": END_OFFSET2,
+        },
+        ...
+        ]
+    }
+    ```
+    where the `offsets` array stores the offset range in the last processed batch per Kafka partition.
+    </TabItem>
+</Tabs>
+
+### Example
+
+Consider a [Delta Lake connector](/connectors/sources/delta) that reports the following completed frontier:
+
+```json
+{
+    "metadata": {
+        "version":10
+    },
+    "ingested_at":"2025-09-20T22:32:32.916501+00:00",
+    "processed_at":"2025-09-20T22:33:34.112410+00:00",
+    "completed_at":"2025-09-20T22:33:34.112414+00:00"
+}
+```
+
+This means that the pipeline has fully processed all Delta table versions up to version 10.
+To determine how far behind the latest version of the table the pipeline lags,
+we have to go outside Feldera to determine the latest table version and its creation timestamp,
```

Comment:
this example shows how one can determine how much the data in an output delta table lags with respect to the data in the input delta table